engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/FilteredTableDataService.java

```diff
@@ -225,11 +206,16 @@ public void handleTableLocationKeyRemoved(
 
         @Override
         public void handleTableLocationKeysUpdate(
-                @NotNull Collection<LiveSupplier<ImmutableTableLocationKey>> addedKeys,
-                @NotNull Collection<LiveSupplier<ImmutableTableLocationKey>> removedKeys) {
+                @NotNull final Collection<LiveSupplier<ImmutableTableLocationKey>> addedKeys,
+                @NotNull final Collection<LiveSupplier<ImmutableTableLocationKey>> removedKeys) {
             final TableLocationProvider.Listener outputListener = getWrapped();
             if (outputListener != null) {
-                outputListener.handleTableLocationKeysUpdate(addedKeys, removedKeys);
+                // Produce filtered lists of added and removed keys.
+                final Collection<LiveSupplier<ImmutableTableLocationKey>> filteredAddedKeys = addedKeys.stream()
```

Comment:
It would be nice to be able to generate these collections only once, rather than once per listener. Doing so would require something of a redesign; we'd probably need to work more like a `SubscriptionAggregator`. Let's comment (without using the word "todo" ;-)) on that optimization, but defer it for now.